File: scripts/explore_dense_result/map_dense_tree.py
import pandas as pd
import matplotlib.pyplot as plt
from Bio import Phylo

TRG_FILE = "/home/eltem/Documents/Cours/M2/Stage/M2_stage_I2BC/data/dense/archive_tree_v1/GCA_001433455@Thermococcus_barophilus_CH5/TRG_match_matrix.tsv"
GENE = "TBCH5v1_RS02920"
TREE = "/home/eltem/Documents/Cours/M2/Stage/M2_stage_I2BC/data/whole_tree.nwk"


def get_trg_line():
    """Get the line corresponding to the gene"""
    trg_df = pd.read_csv(TRG_FILE, sep="\t", index_col=0)
    # get the indexes that have the gene name in their names
    gene_indexes = [index for index in trg_df.index if GENE in index]
    if len(gene_indexes) == 0:
        print("No gene found")
        exit()
    elif len(gene_indexes) > 1:
        print("Multiple genes found")
        exit()
    return trg_df.loc[gene_indexes[0]]


def print_tree(tree_file):
    """Print the tree"""
    tree = Phylo.read(tree_file, "newick")
    Phylo.draw(tree)




if __name__ == "__main__":
    # get info on TRG
    trg_line = get_trg_line()
    # print tree
    print_tree(TREE)
    

File: scripts/explore_dense_result/map_dense_tree.R
library(ggplot2)
library(ggtree)
library(ape)
library(ggnewscale)

trg_file <- "/home/eliott.tempez/Documents/archaea_data/dense/Thermococcus_barophilus_CH1/TRG_match_matrix.tsv"
gene_name <- "TBCH1v1_0272.mRNA.0_elongated"
tree <- "/home/eliott.tempez/Documents/archaea_data/whole_tree.nwk"
focal_species = "Thermococcus_barophilus_CH1"


# Read the TRG file
trg <- read.table(trg_file, header = TRUE, sep = "\t", row.names = 1)
# Keep the row corresponding to the gene of interest
trg <- trg[gene_name, ]
trg <- t(trg)
colnames(trg) <- "type"
rownames(trg) <- gsub("\\.", "@", rownames(trg))
for (i in seq_len(nrow(trg))) {
  if (trg[i, "type"] == "CDS") {
    trg[i, "type"] <- "CDS"
  } else if (substr(trg[i, "type"], 1, 2) == "gS") {
    trg[i, "type"] <- "gS"
  } else if (substr(trg[i, "type"], 1, 3) == "gNS") {
    trg[i, "type"] <- "gNS"
  } else {
    trg[i, "type"] <- "no match"
  }
}


# Read the tree
tree <- read.tree(tree)
# Plot the tree
p <- ggtree(tree, branch.length = "none") +
  geom_tippoint(aes(subset = (label == focal_species)), shape = 23,
                color = "black", fill = "yellow",
                size = 3, stroke = 1, alpha = 1)
p <- p + new_scale_fill()
p <- gheatmap(p, data = trg, offset = 0.1, width = 0.1)
p